packages/overture-schema-core/src/overture/schema/core/parser.py

```diff
@@ -0,0 +1,48 @@
+from typing import Any
+
+from pydantic import BaseModel
+
+from ._cache import get_type_adapter
+
+
+def parse_feature(
+    feature: dict[str, Any], model_type: type[BaseModel] | Any, mode: str = "json"
+) -> dict[str, Any] | None:
+    """
+    Parse and validate a feature using the provided model type.
+
+    Args:
+        feature: Feature data (GeoJSON or flattened format)
+        model_type: Pydantic model type or union type to validate against
+        mode: Output mode - "json" for GeoJSON format, "python" for flattened format
```

Comment:
Can we make an `enum` for `mode`?

Reply:
We could. This is built-in Pydantic behavior and I haven't run across an existing enum.